amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/FlinkKinesisProducer.java

```diff
@@ -301,7 +302,7 @@ public void invoke(OUT value, Context context) throws Exception {
 		}
 
 		ListenableFuture<UserRecordResult> cb = producer.addUserRecord(stream, partition, explicitHashkey, serialized);
-		Futures.addCallback(cb, callback);
+		Futures.addCallback(cb, callback, MoreExecutors.directExecutor());
```

Comment:
The 2-arg version uses `directExecutor()` internally, so this should have the same behaviour